### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by alvingithub1**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅] Coding Challenges
- [✅] Coding Concepts
- [✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

Please advise skill level for this week assessment (incomplete/complete/novice/advanced beginner/competent)

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -22,41 +22,135 @@ const numbersArray2 = [24, 27, 30, 33, 36]
 
 // b) Create the function that makes the test pass.
 
-
+// Canned code below for function tests
```
Comment (elysemontano) [inline code-challenges.test.js:25]: Nice!

File: code-challenges.test.js [hunk 2]
```diff
@@ -22,41 +22,135 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
 // --------------------1) Create a function that takes a number as an argument and decides if the number is evenly divisble by three or not.
 
+// Psuedocode:
+
+// 1. Create/declare a function called divisibleByThree
+// 2. Pass a parameter called number. This number will be used in the function
+// 3. Use an if statement if the number modulo 3 is strictly equal to zero
+    // 3a. Use a return statement that the number is divisible by three
+// 4. Otherwise, use an else statement
+    // 4a. Then use a return statement that the number is not divisible by three
```
Comment (elysemontano) [inline code-challenges.test.js:41]: This is flawless pseudocode!  Super descriptive and to the point, great work!

File: code-challenges.test.js [hunk 3]
```diff
@@ -22,41 +22,135 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+
+
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("divisibleByThree", () => {
+  it("takes a number as an argument and decides if the number is evenly divisble by three or not.", () => {
+    expect(divisibleByThree(15)).toEqual(15 + " is divisible by three")
+    expect(divisibleByThree(0)).toEqual(0 + " is divisible by three")
+    expect(divisibleByThree(-7)).toEqual(-7 + " is not divisible by three")
```
Comment (elysemontano) [inline code-challenges.test.js:50]: I would recommend avoiding using any type of concatenation or interpolation in your test.  The test is meant to just compare an output of your function to a predefined expected output.

File: code-challenges.test.js [hunk 4]
```diff
@@ -22,41 +22,135 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+    expect(arrayCapitalized(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+  })
+})
+
+// //     ReferenceError: arrayCapitalized is not defined
+
+
+// // b) Create the function that makes the test pass.
+
+const arrayCapitalized = (array) => {
```
Comment (elysemontano) [inline code-challenges.test.js:115]: 💯

File: code-challenges.test.js [hunk 5]
```diff
@@ -67,3 +161,24 @@ const vowelTester3 = "challenges"
...
+    for(let i=0; i<vowel.length; i++) {
+        if(vowel[i] === "a" || 
+           vowel[i] === "e" || 
+           vowel[i] === "i" || 
+           vowel[i] === "o" || 
+           vowel[i] === "u") {
+            return i
+        }
+    }
+}
```
Comment (elysemontano) [inline code-challenges.test.js:175]: Excellent work!  If you want to stretch yourself, think of some edge cases as well such as capitalized vowels.

File: code-challenges.test.js [hunk 6]
```diff
@@ -22,41 +22,135 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
-// b) Create the function that makes the test pass.
 
 
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
-// a) Create a test with expect statements for each of the variables provided.
+// Psuedocode:
+
+// I tried using higher order functions, using the .filter method, but gave me an error: TypeError: firstVowel.filter is not a function. I realized that .map/.filter can only be used in arrays. The question is asking about using a string input, so I changed my method to a basic for/if condition.
```
Comment (elysemontano) [inline code-challenges.test.js:131]: In the future, leave that code in here and just comment it out if it doesn't work.  I like to see the process, and perhaps I can help guide you to making it work.

File: coding-concepts.js [hunk 7]
```diff
@@ -67,7 +75,8 @@ class Learn {
...
 }
 var learnStudent = new Learn("George")
-// console.log(learnStudent)
+console.log(learnStudent)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: { this.name = "George", this.cohort = "Bravo", this.year = 2022 }
+// b) Verify and explain: Learn { student: 'George', cohort: 'Bravo', year: 2022 }
+//    The output was pretty close to what I had expected, but the formatting was off. However, the output information was still the same and was "George", "Bravo", and 2022.
```
Comment (elysemontano) [inline coding-concepts.js:82]: Excellent work in this section!!

File: interview-questions.md [hunk 8]
```diff
@@ -8,41 +8,45 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+Your answer: A function requires a return. A function can input parameter values to manipulate data using conditionals and iterating through the input. A method is similar to a function that can manipulate data and return an input.
 
-  Researched answer:
+Researched answer: A function and method are a set of instructions, but a method is associated with an object. Functions are like recipes and execute a set of instructions on data or variables and return the result. A method is similar in that it can perform a task, but the difference is a method is associated with an object and a function is not. Some examples of built-in methods are toLowerCase() and toUpperCase().
```
Comment (elysemontano) [inline interview-questions.md:38]: This answer is starting to look very similar to some documentation.  Make sure that you are forming this in your own words and not rewording documentation.

**Review by elysemontano — APPROVED** [review]
Your work on this assessment is great Alvin!  This assessment was at an Advanced Beginner level (which is great!). 
You have incredible pseudocode, provide great tests and your logic is spot on.  Consider stretching yourself by thinking about edge cases.  
Your researched interview answers have striking similarities to documentation.  Careful with how closely you paraphrase.  When researching, I highly recommend trying to come up with your own way to phrase the concepts you read about.
